testpar/t_bigio.c

```diff
@@ -1092,11 +1092,24 @@ single_rank_independent_io(void)
         file_id = H5Fcreate(FILENAME[1], H5F_ACC_TRUNC, H5P_DEFAULT, fapl_id);
         VRFY_G((file_id >= 0), "H5Dcreate2 succeeded");
 
-        /*
-         * Calculate the number of elements needed to exceed
-         * MPI's INT_MAX limitation
-         */
-        dims[0] = (INT_MAX / sizeof(int)) + 10;
+        if (sizeof(size_t) > 4) {
+            /*
+             * Calculate the number of elements needed to exceed
+             * MPI's INT_MAX limitation
+             */
+            dims[0] = (INT_MAX / sizeof(int)) + 10;
+        }
+        else {
+            /* 32-bit systems can't allocate a buffer > MPI's INT_MAX
+             * limitation, so that aspect of this test can't really
+             * be tested. Instead, just get enough elements to fill
+             * a 2 GiB buffer, which should be safe on most platforms
+             * (even though 4 GiB technically fits in the data type,
+             * the operating system's allocator probably can't fulfil
+             * a request that large).
+             */
+            dims[0] = ((INT_MAX / 2) / sizeof(int));
```

Comment:
Good fix, though this entire test is largely useless on 32-bit systems since it won't really test anything that other tests don't.

Reply:
I should also note that's basically the same case for t_2gio, so we should really decide if running it on 32-bit systems even makes sense, as it would basically be defeating the purpose of the test.

Reply:
Agree. I thought about just skipping it, but testing the 2 GiB barrier is possibly useful.

Reply:
I'm assuming INT_MAX will be 2147483647 most places, in which case wouldn't this be grabbing ~1GiB of elements instead?

Reply:
I'll look at t_2gio tonight and think about it

Reply:
t_2Gio passes on my 32-bit Debian 12 VM w/ the develop branch

Reply:
I think the t_2Gio test really needs rethought anyway. It first sets the "big I/O size" to 1MiB then writes only ~128MiB datasets, in which case it's not actually testing anything about 2GiB of I/O, or the MPI int limitation, but rather just whether the derived datatype creation functionality in the library works as expected. Did you happen to try running the test with a testexpress level of 0? Either set it to 0 at configure time or run the test with the `-testexpress 0` command-line argument. That at least runs a test that's supposed to test 4GiB I/O.

Reply:
I'll give it a try tonight